fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -1522,33 +1526,49 @@ public FHIRPersistenceTransaction getTransaction() {
     public boolean isDeleteSupported() {
         return true;
     }
-    
+
     private FHIRDbDAO getBaseDao() {
         return baseDao;
     }
-    
+
     private void setBaseDao(FHIRDbDAO baseDao) {
         this.baseDao = baseDao;
     }
-    
+
     private Connection getManagedConnection() {
         return managedConnection;
     }
 
     private void setManagedConnection(Connection managedConnection) {
         this.managedConnection = managedConnection;
     }
-    
+
     private OperationOutcome buildOKOperationOutcome() {
         return FHIRUtil.buildOperationOutcome("All OK", IssueType.INFORMATIONAL, IssueSeverity.INFORMATION);
     }
 
     private OperationOutcome buildErrorOperationOutcome() {
         return FHIRUtil.buildOperationOutcome("The database connection was not valid", IssueType.NO_STORE, IssueSeverity.ERROR);
     }
-    
+
     private Connection createConnection() throws FHIRPersistenceDBConnectException {
         FHIRDbDAOImpl dao = new FHIRDbDAOImpl();
         return dao.getConnection();
     }
+
+    @Override
+    public void closeConnection() throws FHIRPersistenceException {
```

Comment:
I personally don't think this is the right place or the correct change.  If you are getting it from a pool, the pool should manage the closing of the connection in this case.  In particular, this is coming from the `jdbc/fhirProxyDataSource` ... I think this has unintended consequences.

Reply:
The same in our commit method which closes the sharedConnection, I believe the close() is just return the connection to the connection pool instead of closing the real connection to the DB.